driver/src/main/java/oracle/nosql/driver/NoSQLHandleConfig.java

```diff
@@ -553,6 +561,23 @@ public int getDefaultRequestTimeout() {
         return timeout == 0 ? DEFAULT_TIMEOUT : timeout;
     }
 
+    /**
+     *
+     * @return Http protocol settings
+     */
+    public List<String> getHttpProtocols() {
+        return httpProtocols;
+    }
+
+    /**
+     * Check if "h2" is in the protocols list
+     *
+     * @return true if "h2" is in the protocols list
+     */
+    public boolean useHttp2() {
```

Comment:
I was thinking that the list would be ordered by preference. While http1.1, http2 doesn't necessarily make sense that would be the logic

Reply:
The list is order irrelevant.
In the ALPN case, if we provide both H2 and H1.1 (no matter what order), the server always pick H2 over H1.1 given the server supports H2.
In the clear text case, when we provide both H2 and H1.1 to the config, it's not like we try H2 first for 30 seconds timeout then fall back to H1.1, instead we directly use H2C protocol, so the order of the protocols is not relevant either.
For the visibility, sure I will remove this method, and do the logic in the lower level code.